scrapy/utils/log.py

```diff
@@ -232,18 +230,10 @@ def logformatter_adapter(logkws: dict) -> Tuple[int, str, dict]:
     and adapts it into a tuple of positional arguments for logger.log calls,
     handling backward compatibility as well.
     """
-    if not {"level", "msg", "args"} <= set(logkws):
-        warnings.warn("Missing keys in LogFormatter method", ScrapyDeprecationWarning)
-
-    if "format" in logkws:
-        warnings.warn(
-            "`format` key in LogFormatter methods has been "
-            "deprecated, use `msg` instead",
-            ScrapyDeprecationWarning,
-        )
 
     level = logkws.get("level", logging.INFO)
-    message = logkws.get("format", logkws.get("msg"))
+    message = logkws.get("msg", "")
+    assert message is not None
```

Comment:
OK, now that I look at the warning, I see 2 things:
- The warning was about `msg` being a required key, so I think `logkws["msg"]` instead of `logkws.get("msg", "")` makes sense now that the warning is gone.
- There was no warning if `logkws["msg"]` was `None`, so we should assume that as a possibility since this dict seems to be user-provided. So we cannot `assert`, we have to fall back: `message = logkws["msg"] or ""`.

Reply:
Hi Gallarcio,
I don't really understand why we have to fall back: `message = logkws["msg"] or ""`.

Firstly ,this part of the warnings has been removed according to   @wRAR 's comments https://github.com/scrapy/scrapy/pull/6116#discussion_r1363487495
``` python
if not {"level", "msg", "args"} <= set(logkws):
        warnings.warn("Missing keys in LogFormatter method", ScrapyDeprecationWarning)
```
----------------
Secondly:
> [Gallaecio](https://github.com/Gallaecio): The warning was about msg being a required key, so I think logkws["msg"] instead of logkws.get("msg", "") makes sense now that the warning is gone.

Although msg is a requried key, but it is only mentioned and constrained in comments in `class LogFormatter`,  there is no actual code constraints. The user can still provide a dictionary without the `msg key`. Hence, if `key msg` is missing, simply using `logkws["msg"]` would arise a KeyError.

And I agree the point we cannot assert `message is not None`. 
In fact, our current logging system could cope with the case `{"msg": None}`. So maybe only `message = logkws.get("msg", "")` should be fine.

Reply:
I think raising `KeyError` is fine. I think it is the natural evolution from the warning: first msg was probably not necessary, then it became necessary but it worked without it causing a warning, now it will raise KeyError if missing.

If `"msg": None` is OK, then I agree the fall back is not needed, and we can simply use `logkws["msg"]`.

Reply:
Probably the best solution for this is would be:

```python
message = logkws.get("msg") or ""
```

Reply:
Unless like @Gallaecio said we want to generate a exception so the user can fix their code, then we don’t need to ` `get` at all.

Reply:
Hi @wRAR, could you please review and merge this pull request?
Thanks!